File: Project Nothing/Server/src/protocol/client/ServerEcho.cpp
#include "stdafx.h"

#include "network/CNetProtocol.h"

#include "ServerEcho.h"

ServerEcho::ServerEcho ()
	: m_nNum (0)
{
}

ServerEcho::~ServerEcho ()
{
}

void ServerEcho::serialize (COutStream& _kOut_Stream)
{
}

void ServerEcho::deserialize (CInStream& _kIn_Stream)
{
	_kIn_Stream >> m_nNum;
}

void ServerEcho::excute ()
{
}

File: Unity/Project Nothing/Assets/Script/Protocol/Server/ServerEcho.cs
using ProjectNothing.Network;

namespace ProjectNothing.Protocol
{
    public sealed class ServerEcho : NetProtocol<ServerEcho>
    {
        public string m_String;

        public override void Deserialize (BitInStream inStream)
        {
        }

        public override void Excute ()
        {
        }

        public override void Serialize (BitOutStream outStream)
        {
            outStream.Write (m_String);
        }
    }
}
